### ISSUE TITLE
Feature Request: Energizing Orb only accepts one set of items

### ISSUE THREAD

**Issue by Eydamos** [zeroneye/Powah#108]
#### Version: (make sure you are on the latest version before reporting):

- Minecraft: 1.16.1
- Forge: 32.0.108
- Powah: 2.2.0

#### Feature description:
An Energizing Orb can only craft on set of an item. E.g. it can transform one diamond into one Niotic Crystal. But if you put a hopper next to it with a stack of diamonds it will put 6 diamons in it even though it can not craft more than one Niotic Crystal at a time.
My request would be that if one diamond is in the orb it will not accept another diamond or any other item.
This should of course be recipe aware so if you want to craft Energized Steel it will accept one Iron Ingot and accept one Gold Ingot but not any other items or another Iron or Gold Ingot.

Test Case 1:
Two Hoppers pointing towards the Energizing Orb. One Hopper with Diamonds one Hopper with Emeralds. The block should only accept items of one of the Hoppers. If one Hopper is empty it starts accepting items from the other one.

Test Case 2:
Two Hoppers pointint towars the Energizing Orb. One Hopper with Iron Ingots and Gold Ingots and one Hopper with Diamonds.
The Energizing Orb will either accept the Diamonds until the hopper is empty or an Iron or Gold Ingot until those are empty and then it starts accepting the Diamonds.
Yes I am aware that if I would have an uneven amount of Iron and Gold Ingots in the Hopper this would block the Diamonds and any crafting until I put enough Ingots of the other type into the Hopper to balance this out.

**Comment by tecnobrat** [comment]
I think this would also fix the issue with autocrafting with refined storage inserting too many items at once and making autocrafting not really viable.

**Comment by andrewsf** [comment]
Are you _sure_ autocrafting is not viable?

![image](https://user-images.githubusercontent.com/697565/100030072-850de200-2da7-11eb-85c5-1f31df054dd2.png)
![image](https://user-images.githubusercontent.com/697565/100038940-3b2df780-2db9-11eb-9e07-5bd3cf30f79d.png)
![image](https://user-images.githubusercontent.com/697565/100039052-5bf64d00-2db9-11eb-85fd-58cffe09aa2d.png)
![image](https://user-images.githubusercontent.com/697565/100039522-3cabef80-2dba-11eb-9182-446cc4058b38.png)
![image](https://user-images.githubusercontent.com/697565/100039261-b68fa900-2db9-11eb-89b0-65fe7d64b982.png)

**Comment by TheVekter** [comment]
Viable? Maybe. Way more of a pain in the butt than it needs to be? Definitely.

**Comment by JackTWing** [comment]
how do you craft nitro crystal?  i have spent 2 hours looking for the recipe

**Comment by TheVekter** [comment]
I believe it's a nether star, two redstone blocks, and a block of blazing
crystals but it might be different depending on the pack. JEI should
support it.

**Comment by andrewsf** [comment]
> Viable? Maybe. Way more of a pain in the butt than it needs to be? Definitely.

Obviously the charging mechanic of this mod is designed to challenge your brain a little bit; fortunately for the disinclined I already solved all the hard problems (which were simpler than many vanilla farm problems). It should take less than 20 minutes to build an autocrafting setup matching my screenshots.

Parts list to autocraft something w/ AE2: 2 interfaces, 2 redstone dust, 2 comparators, 2 solid blocks, 1 composter, 1 hopper. And of course some charging lasers.

Build one setup for each number of ingredients (so all 2-ingredient recipes can share the same setup, etc.). Fill the composter to an appropriate level for the number of ingredients.

I find this solution to be very cheap and easy to build, considering that it solves one of the most difficult challenges of this mod. I do not agree that this is a "pain in the butt" or that the mod needs to make it even easier than this.

**Comment by TheInfamousAlk** [comment]
I disagree with the above statement. I think that "early game" it should be a pain to automate, but once you get into late game with full AE2 autocrafting, etc, it should be easy but complex to automate. In my (and most people's) automation rooms, you'll have entire walls filled with machines that all have ae2 interfaces on them and rows of cables, all neatly organized, and then you have the "powah section" that is just a huge jumble of redstone, hoppers and COMPOST BINS. It's unnecessarily complicated in all the wrong ways. 

I like complicated mods. But when you add complication via having to use weird out-of-theme methods instead of adding content that works together in the mod, it's not fun. For example, my mekanism reactor is surrounded by a sea of machines that are all connected together with pipes and cables all to produce one substance to fuel the reactor. That's fun to set up. But having to put so much effort into autocrafting such insignificant items and having to use vanilla hoppers (which are notoriously laggy) and comparators and compost bins, (and this is the most important part of my whole post) it makes me and I would imagine a majority of people simply not want to use the mod. 

It's a super fun mod, and very useful, but this one problem has made me stop playing because other mods are able to be integrated more elegantly into various systems. 

Now you could say "if you don't like it, play other mods. it wasn't made for people who don't like this mechanic" which would be fair if we were talking about me not liking the complexity in general, and it is fair with something like AE2 vs refined storage. AE2 and RS do basically the same thing, but AE2 has so many more complicated restrictions. I personally play AE2 because the extra mechanics are fun. I enjoy managing channels and building a controller and all that stuff. But that's the thing, AE2's arbitrary complications are additions to the mod that are fun to look at, use, and learn. This powah "arbitrary complication" is not an addition to the mod. It's simply a small block behavior that throws a wrench in otherwise universal mod automation. It requires out-of-mod solutions that aren't intuitive or fun to look at, use and learn. 

So what I propose is that Powah add additional layers of complexity to the charging/crafting system, but make all of those accessible out of the box to things like pipes, exporters, importers, etc. Such as having the chargers have lenses that are different for each craft, or specific items in the chargers that get consumed and "combined" with the orb, or maybe a multi-orb setup where they all get combined. The possibilities are endless. But no matter what, I think the specific step of "inserting an item" should be extremely easy. The complexity should come from elsewhere. You can have an infinitely complex mod (like Create for example) where each individual interaction and block behavior is simple and doesn't require silly setups to hook into other mods. The complexity  and challenge comes from the number of interactions and how you can most efficiently tie them in to each other. 

Or in other words, I enjoy spending large amounts of effort on solving large problems, not medium amounts of effort solving very very small problems. Once you get to endgame in most big modpacks, you're now solving large-scale problems. Dealing with tiny annoying problems like this becomes annoying (especially at scale).

**Comment by andrewsf** [comment]
_Note: I have never worked on this mod and do not speak for the mod's design or future in any way._

> a huge jumble of redstone, hoppers and COMPOST BINS
> having to use vanilla hoppers (which are notoriously laggy) and comparators and compost bins

I respect that some may find my solution smelly according to their personal preferences. The design in my screenshot was chosen specifically (and carefully refined) to show that at least one solution exists wherein:

1. no mods are required other than Powah and an autocrafting mod
1. very few & very cheap blocks are required
1. very little space is required (smaller than 5x2x2 including interfaces)
1. basic vanilla mechanics are sufficient — comparing 2 signals and locking a hopper. Naturally if you don't like composters and comparators for generating signals, or if you don't like redstone dust, or if you hate hoppers, or if you want mod-only solutions for some reason, you absolutely could make use of one of the myriad mods that cram all that functionality into one block.

I am well aware people want to make their own solutions, so my goal was to showcase it is broadly possible in _all_ modpacks with zero changes to the Powah mod. If copy/pasting my exact design unblocks some impatient or exasperated people, great. If it inspires players to implement something they like way better, perfect.

> For example, my mekanism reactor is surrounded by a sea of machines that are all connected together with pipes and cables all to produce one substance to fuel the reactor.

All of my Mekanism reactors, the SPS, and waste disposal were connected in a single vanilla redstone system to handle all possible types of faults and ensure the base's battery reserves always stay between 80% and 98% full. It was a fun problem to solve and in fact was more fun than any planned problem Mekanism threw at me. I don't agree at all with the _apparent_ suggestion that redstone is out of place in the middle of other mods. On the contrary I believe tech mods are at their best when they integrate via common and clear interfaces, and I don't see one more common and clear than redstone.

Personally I don't mind if Powah makes auto-energizing easier. It's not my mod, and my modded playthrough reached its conclusion. But after appreciating in my own play that there was one actual challenge in this mod, I was really rubbed the wrong way by the dramatic attitudes I kept seeing around this particular topic and the sheer resignation from some players. So I posted my solution so that frustrated players can know with 100% certainty that there are solutions available, today. Hopefully that's enough for some.

> I enjoy spending large amounts of effort on solving large problems

Food for thought: Making a large and complex machine that is both robust and attractive at the same time is a large problem that can require large effort. And it can feel good: watching modpack playthroughs, many players seem to find it more rewarding than configuring I/O on a hundred -ificator blocks. Just my impression.

**Comment by hakanai** [comment]
I'd be happy if #199 were fixed because that alone would make AE2 autocrafting less annoying.

Being able to stack more than one item in would be nice in its own way for earlier game, but it seems kind of unreasonable to allow multiple items from multiple recipes at the same time as you would then have a lot more code to keep track of which sub-recipe has how much power, a complication you don't have at all if you only allow items of a single recipe to mix. And later game you're going to want to automate more than one recipe, and having multiple orbs is awkward.

But the interface's blocking mode not even working is what really bugs me when trying to automate this. I'm sure I could build a bigger automation system for it and it would work, but having to move the entire setup was something I was trying to avoid.
